### ISSUE TITLE
Discriminator loss plateau on DIV2K

### ISSUE THREAD

**Issue by EKami** [leftthomas/SRGAN#3]
Hello,
I'm not a GAN expert but I tried your code on the [DIV2K](https://s3-eu-west-1.amazonaws.com/torchlight-datasets/DIV2K.zip) dataset (I uploaded it on my S3 account to be easier to download) and after few epochs the discriminator loss seems to stop decreasing (which I believe means it stops giving relevant feedback to the generator). Here are the logs:
```
[12/100] Loss_D: 1.0234 Loss_G: 0.0090 D(x): 0.6659 D(G(z)): 0.6893: 100%|██████████████| 13/13 [00:15<00:00,  1.22s/it]
[converting LR images to SR images] PSNR: 22.3313 dB SSIM: 0.6381: 100%|██████████████| 100/100 [00:33<00:00,  3.03it/s]
[saving training results]: 100%|████████████████████████████████████████████████████████| 20/20 [00:14<00:00,  1.40it/s]
[13/100] Loss_D: 1.0124 Loss_G: 0.0082 D(x): 0.6671 D(G(z)): 0.6795: 100%|██████████████| 13/13 [00:16<00:00,  1.26s/it]
[converting LR images to SR images] PSNR: 22.1826 dB SSIM: 0.6464: 100%|██████████████| 100/100 [00:32<00:00,  3.04it/s]
[saving training results]: 100%|████████████████████████████████████████████████████████| 20/20 [00:14<00:00,  1.42it/s]
[14/100] Loss_D: 1.0271 Loss_G: 0.0091 D(x): 0.6130 D(G(z)): 0.6401: 100%|██████████████| 13/13 [00:16<00:00,  1.23s/it]
[converting LR images to SR images] PSNR: 21.5150 dB SSIM: 0.6378: 100%|██████████████| 100/100 [00:33<00:00,  3.03it/s]
[saving training results]: 100%|████████████████████████████████████████████████████████| 20/20 [00:14<00:00,  1.41it/s]
...
[97/100] Loss_D: 1.0000 Loss_G: 0.0044 D(x): 1.0000 D(G(z)): 1.0000: 100%|██████████████| 13/13 [00:16<00:00,  1.26s/it]
[converting LR images to SR images] PSNR: 24.2824 dB SSIM: 0.7469: 100%|██████████████| 100/100 [00:33<00:00,  3.03it/s]
[saving training results]: 100%|████████████████████████████████████████████████████████| 20/20 [00:13<00:00,  1.44it/s]
[98/100] Loss_D: 1.0000 Loss_G: 0.0044 D(x): 1.0000 D(G(z)): 1.0000: 100%|██████████████| 13/13 [00:16<00:00,  1.27s/it]
[converting LR images to SR images] PSNR: 22.9190 dB SSIM: 0.7432: 100%|██████████████| 100/100 [00:33<00:00,  3.01it/s]
[saving training results]: 100%|████████████████████████████████████████████████████████| 20/20 [00:13<00:00,  1.44it/s]
[99/100] Loss_D: 1.0000 Loss_G: 0.0047 D(x): 1.0000 D(G(z)): 1.0000: 100%|██████████████| 13/13 [00:16<00:00,  1.27s/it]
[converting LR images to SR images] PSNR: 24.6174 dB SSIM: 0.7501: 100%|██████████████| 100/100 [00:33<00:00,  3.03it/s]
[saving training results]: 100%|████████████████████████████████████████████████████████| 20/20 [00:13<00:00,  1.44it/s]
[100/100] Loss_D: 1.0000 Loss_G: 0.0044 D(x): 1.0000 D(G(z)): 1.0000: 100%|█████████████| 13/13 [00:16<00:00,  1.27s/it]
[converting LR images to SR images] PSNR: 24.8851 dB SSIM: 0.7500: 100%|██████████████| 100/100 [00:32<00:00,  3.04it/s]
```
At some point the SSIM metric will not improve above ~0.75 and the PSNR above ~24/25.
Here is an example of an image augmented with the `test_image.py` script after training for 100 epochs:

# The original image
![img_1190](https://user-images.githubusercontent.com/4030626/36342047-a5e47e4c-13f8-11e8-970a-9f107de4996e.JPG)

# The SR image
![out_srf_4_img_1190](https://user-images.githubusercontent.com/4030626/36342051-b7b7a43c-13f8-11e8-8f76-ab23d978be40.JPG)

I don't really see any difference between the two.
Is this a normal behavior? Thank you.

**Comment by leftthomas** [comment]
@EKami I suggest you change the loss function and retrain again, because the loss of my code is different with the paper, and in the training time, you should look the valing results to see if it stop imporving, not just the loss value.
